src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MD5.cs

```diff
@@ -112,6 +115,133 @@ public static bool TryHashData(ReadOnlySpan<byte> source, Span<byte> destination
+        ///   The default value is <see cref="System.Threading.CancellationToken.None" />.
+        /// </param>
+        /// <returns>The hash of the data.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="source" /> does not support reading.
+        /// </exception>
+        public static ValueTask<byte[]> HashDataAsync(Stream source, CancellationToken cancellationToken = default)
+        {
+            if (source is null)
+                return ValueTask.FromException<byte[]>(new ArgumentNullException(nameof(source)));
+
+            if (!source.CanRead)
+                return ValueTask.FromException<byte[]>(
+                    new ArgumentException(SR.Argument_StreamNotReadable, nameof(source)));
+
+            return LiteHashProvider.HashStreamAsync(HashAlgorithmNames.MD5, HashSizeInBytes, source, cancellationToken);
+        }
+
+        /// <summary>
+        /// Asynchronously computes the hash of a stream using the MD5 algorithm.
+        /// </summary>
+        /// <param name="source">The stream to hash.</param>
+        /// <param name="destination">The buffer to receive the hash value.</param>
+        /// <param name="cancellationToken">
+        ///   The token to monitor for cancellation requests.
+        ///   The default value is <see cref="System.Threading.CancellationToken.None" />.
+        /// </param>
+        /// <returns>The total number of bytes written to <paramref name="destination" />.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <p>
+        ///   The buffer in <paramref name="destination"/> is too small to hold the calculated hash
+        ///   size. The MD5 algorithm always produces a 128-bit hash, or 16 bytes.
+        ///   </p>
+        ///   <p>-or-</p>
+        ///   <p>
+        ///   <paramref name="source" /> does not support reading.
+        ///   </p>
+        /// </exception>
+        public static ValueTask<int> HashDataAsync(
+            Stream source,
+            Memory<byte> destination,
+            CancellationToken cancellationToken = default)
+        {
+            if (source is null)
+                return ValueTask.FromException<int>(new ArgumentNullException(nameof(source)));
+
+            if (destination.Length < HashSizeInBytes)
+                return ValueTask.FromException<int>(
+                    new ArgumentException(SR.Argument_DestinationTooShort, nameof(destination)));
+
+            if (!source.CanRead)
+                return ValueTask.FromException<int>(
+                    new ArgumentException(SR.Argument_StreamNotReadable, nameof(source)));
```

Comment:
throw :smile:

Reply:
🤔 thought I fixed all of these from Steve's feedback. Maybe I didn't push something.

Reply:
Looks like only MD5 got missed.  Probably just missed it as the only hash to not start with S.

Reply:
Ah I missed MD5 because I'm a dummy.